src/hooks/useSavingsAccount.js

```diff
@@ -105,13 +106,16 @@ function usePersistentBackupSavings(accountAddress, network) {
 
 export default function useSavingsAccount(includeDefaultDai) {
   const dispatch = useDispatch();
+  const { selectedWallet } = useWallets();
   const { accountAddress, network } = useAccountSettings();
   const [backupSavings = null, setBackupSavings] = usePersistentBackupSavings(
     accountAddress,
     network
   );
 
-  const hasAccountAddress = !!accountAddress;
+  const hasAccountAddress =
+    !!accountAddress &&
+    selectedWallet.addresses.find(({ address }) => accountAddress === address);
```

Comment:
were you seeing long delays / discrepancies in `accountAddress` and the `selectedWallet`?
since a user can switch between multiple accounts underneath a selected wallet (in which case this would return `true`), this isn't a very reliable mechanism. we might want to bring in some other flag that indicates loading in order for the query to skip

Reply:
hey @jinchung, yeah good point. I amended the PR and made it so that the "Savings" & "Pools" headers don't display while the assets are loading. I think this may be better UX?